From: Jan Blunck <jblunck@suse.de>
Subject: Dump elf headers to core per default

This patch changes the default of /proc/self/coredump_filter to write the elf
header of shared objects and the executable itself to the core file. This
change is necessary to read the build-ids generated by GCC from the produced
core files. These core files might not be readable by anncient versions of
GDB.

Signed-off-by: Jan Blunck <jblunck@suse.de>
---
 include/linux/sched.h |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: b/include/linux/sched.h
===================================================================
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -407,8 +407,9 @@ extern int get_dumpable(struct mm_struct
 #define MMF_DUMP_FILTER_BITS	5
 #define MMF_DUMP_FILTER_MASK \
 	(((1 << MMF_DUMP_FILTER_BITS) - 1) << MMF_DUMP_FILTER_SHIFT)
-#define MMF_DUMP_FILTER_DEFAULT \
-	((1 << MMF_DUMP_ANON_PRIVATE) |	(1 << MMF_DUMP_ANON_SHARED))
+#define MMF_DUMP_FILTER_DEFAULT						\
+	((1 << MMF_DUMP_ANON_PRIVATE) |	(1 << MMF_DUMP_ANON_SHARED) |	\
+	 (1 << MMF_DUMP_ELF_HEADERS))
 
 struct sighand_struct {
 	atomic_t		count;
